src/lib/logger.ts

```diff
@@ -0,0 +1,19 @@
+import winston, { format } from 'winston'
+
+const { combine, timestamp, prettyPrint } = format
+
+export const createBaseLogger = (path: string) => {
+  const formats = [
+    winston.format.json(),
+    timestamp(),
+    process.env.NODE_ENV === 'development' ? prettyPrint() : undefined,
+  ].filter((v) => !!v) as winston.Logform.Format[]
+
+  return winston.createLogger({
+    format: combine(...formats),
```

Comment:
may want to [prevent exit on handling uncaught exceptions](https://github.com/winstonjs/winston#to-exit-or-not-to-exit) (not sure if it applies to nextjs, but better be safe than sorry?)

```ts
  exitOnError: false
```